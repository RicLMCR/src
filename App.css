.App {
  text-align: center;
  max-width: 100vw;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.Nav-bar {
  background-color: #282c34;
  height: 100px;
  display: flex;
  flex-direction: row;
  /* align-items: center;
  justify-content: center; */
  color: white;
  width: 100%;
  /* border: blue solid 2px; */
}

.Nav-bar p {
  font-size: 20px;
  margin-right: 10%;
  /* border: red solid 2px; */
  height: 30%;
  cursor: pointer;
}

.h1-nav {
  color: #61dafb;
  float: left;
  margin-right: 10px;
  font-size: 25px;
  /* margin-top: 45px; */
  /* border: red solid 2px; */
  width: 20%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.input-nav {
  float: right;
  padding: 5px 15px;
  background: #ccc;
  border: 0 none;
  cursor: pointer;
  -webkit-border-radius: 5px;
  border-radius: 5px;
  border-color: black;
}

.button-nav {
  float: right;
  background-color: #06eded94;
  width: 170px;
  border-radius: 5px;
  border-color: black;
}

.form-group {
  margin-bottom: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-right: 70px;
}

.logo-nav {
  max-width: 250px;
  max-height: 250px;
  /* margin-right: 10vw; */
}

.modal {
  /* border: red solid 2px; */
  position: absolute;
  right: 5%;
  top: 20%;
}

.img-div {
  /* border: solid red 2px; */
  margin: 0;
  padding: 0;
  display: flex;
  width: 30vw;
  justify-content: flex-start;
}
.signInContainer {
  /* border: red solid 2px; */
  width: 10%;
  display: flex;
  justify-content: flex-end;
  align-items: center;
}
