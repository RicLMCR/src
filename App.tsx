import React, { useEffect } from "react";

import "./App.css";
import Navbar from "./components/navBar/Navbar";
import { useState } from "react";
import UserInfo from "./components/userInfo/UserInfo";
import BookTO from "./components/bookTO/BookTO";

function App() {
  //user information on login
  const [data, setData] = useState<any>("");

  //booking information
  const [datesBooked, setDatesBooked] = useState<any>([]);
  const [dateOne, setDateOne] = useState<Date>();
  const [dateTwo, setDateTwo] = useState<Date>();
  const [bookBoolean, setBookBoolean] = useState<Boolean>(false);
  const [getDay, setGetDay] = useState<string>("");
  const [userBookings, setUserBookings] = useState<any>();
  const [loading, setLoading] = useState<boolean>(false);

  //  console.log(datesBooked);
  useEffect(() => { }, [datesBooked]);

  return (
    <div className="App">
      <Navbar setData={setData} data={data} setUserBookings={setUserBookings} userBookings={userBookings} setLoading={setLoading} />
      <div className="bodyContent">
        <UserInfo data={data} userBookings={userBookings} loading={loading} />
        {data ? (
          <BookTO
            dateOne={dateOne!}
            setDateOne={setDateOne}
            dateTwo={dateTwo!}
            setDateTwo={setDateTwo}
            datesBooked={datesBooked}
            setDatesBooked={setDatesBooked}
            bookBoolean={bookBoolean}
            setBookBoolean={setBookBoolean}
            data={data}
            setGetDay={setGetDay}
            getDay={getDay}
            loading={loading}
          />
        ) : null}
      </div>
    </div>
  );
}

export default App;
