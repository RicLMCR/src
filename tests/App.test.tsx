<<<<<<< HEAD
import App from "../App";
import renderer from "react-test-renderer";

describe("Tests that the app its self renders correctly", () => {
  it("App renders correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
=======
import App from '../App';
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react';

describe('Tests that the app its self renders correctly', () => {

  it(' App renders correctly', () => {
    render(<App />)
    //  .toJSON()
    // expect(tree).toMatchSnapshot()
  })
})
>>>>>>> e94590af27696d5f49242531a491a8365cd6e3e5
