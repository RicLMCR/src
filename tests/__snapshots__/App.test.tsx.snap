// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests that the app its self renders correctly App renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="Nav-bar"
  >
    <div
      className="img-div"
    >
      <img
        alt="logo"
        className="logo-nav"
      />
    </div>
    <h2
      className="h1-nav"
    >
      Paid Time Off 
    </h2>
    <h2
      className="h1-nav"
    >
      Logged in: 
    </h2>
    <div
      className="signInContainer"
    >
      <p
        onClick={[Function]}
      >
        Sign In
      </p>
    </div>
  </div>
  <div
    className="bodyContent"
  >
    <div
      className="userInfoContainer"
    >
      <div>
         
        <h2>
          Sign in
        </h2>
      </div>
      <input
        className="button-nav"
        onClick={[Function]}
        type="submit"
        value="Holidays"
      />
    </div>
  </div>
</div>
`;
